### ISSUE TITLE
Python 3.5.2 build 3 does not place compiled Python files in __pycache__ directory

### ISSUE THREAD

**Issue by jjhelmus** [conda-forge/python-feedstock#70]
Ths conda-forge python 3.5.2_3 packages do not have the `.pyc` files in the standard library in `__pycache__` directory, rather they placed next to the `.py` directory.  The 3.5.2_2 build has the in the correct location.  

For example:

```
$ tar tvf ~/Downloads/osx-64-python-3.5.2-2.tar.bz2 | grep zipfile.*pyc
-rw-r--r--  0 travis staff    5000 Jul 25 20:38 lib/python3.5/test/__pycache__/test_zipfile64.cpython-35.opt-1.pyc
-rw-r--r--  0 travis staff    5000 Jul 25 20:38 lib/python3.5/test/__pycache__/test_zipfile64.cpython-35.pyc
-rw-r--r--  0 travis staff    5000 Jul 25 20:38 lib/python3.5/test/__pycache__/test_zipfile64.cpython-35.opt-2.pyc
-rw-r--r--  0 travis staff   44344 Jul 25 20:38 lib/python3.5/__pycache__/zipfile.cpython-35.opt-2.pyc
-rw-r--r--  0 travis staff   49866 Jul 25 20:38 lib/python3.5/__pycache__/zipfile.cpython-35.opt-1.pyc
-rw-r--r--  0 travis staff   49948 Jul 25 20:38 lib/python3.5/__pycache__/zipfile.cpython-35.pyc
-rw-r--r--  0 travis staff   75071 Jul 25 20:38 lib/python3.5/test/__pycache__/test_zipfile.cpython-35.opt-2.pyc
-rw-r--r--  0 travis staff   77015 Jul 25 20:38 lib/python3.5/test/__pycache__/test_zipfile.cpython-35.pyc
-rw-r--r--  0 travis staff   77015 Jul 25 20:38 lib/python3.5/test/__pycache__/test_zipfile.cpython-35.opt-1.pyc
```

vs.

```
$ tar tvf ~/Downloads/osx-64-python-3.5.2-3.tar.bz2 | grep zipfile.*pyc
-rw-r--r--  0 travis staff    4968 Sep  8 09:39 lib/python3.5/test/test_zipfile64.pyc
-rw-r--r--  0 travis staff    5044 Sep  8 09:38 lib/python3.5/test/__pycache__/test_zipfile64.cpython-35.opt-2.pyc
-rw-r--r--  0 travis staff    5044 Sep  8 09:37 lib/python3.5/test/__pycache__/test_zipfile64.cpython-35.opt-1.pyc
-rw-r--r--  0 travis staff   44388 Sep  8 09:38 lib/python3.5/__pycache__/zipfile.cpython-35.opt-2.pyc
-rw-r--r--  0 travis staff   49910 Sep  8 09:37 lib/python3.5/__pycache__/zipfile.cpython-35.opt-1.pyc
-rw-r--r--  0 travis staff   49911 Sep  8 09:39 lib/python3.5/zipfile.pyc
-rw-r--r--  0 travis staff   75115 Sep  8 09:38 lib/python3.5/test/__pycache__/test_zipfile.cpython-35.opt-2.pyc
-rw-r--r--  0 travis staff   76983 Sep  8 09:39 lib/python3.5/test/test_zipfile.pyc
-rw-r--r--  0 travis staff   77059 Sep  8 09:37 lib/python3.5/test/__pycache__/test_zipfile.cpython-35.opt-1.pyc
```

Unfortunately this results in these `.pyc` files being included in the correct locations in any conda package which are build with this python build in the build environment.  This can bloat the package side by ~900 kB.

I believe this is just a flux of the particular version of conda-build used the build.  Rolling the build number should fix this.  I will investigate.

**Comment by msarahan** [comment]
Don't look too hard.  This is a "feature" on conda-build 1.21.14 and 2.0.0.  It is removed in 2.0.1.

**Comment by jjhelmus** [comment]
Good to know, will try to get the patch for conda-forge-build-setup out so conda-forge can start using conda-build 2.x.  Then this can be fixed.

**Comment by jjhelmus** [comment]
We could also pull these build from anaconda.org.  They are identical to the `_2` except for a change in the `meta.yml` file in the recipe directory to export the `python_branding` environment variable.

**Comment by jakirkham** [comment]
So I think we just need a build number bump at this point, correct?

**Comment by jjhelmus** [comment]
> So I think we just need a build number bump at this point, correct?

Yes, although my guess is that the OS X build will fail with the long prefix.  I'll submit a PR shortly.

**Comment by jjhelmus** [comment]
closed by python 3.5.2 build 4 created by PR #105 

```
$ tar tvf osx-64-python-3.5.2-4.tar.bz2 | grep zipfile.*pyc
-rw-rw-r-- travis/staff   4968 2017-01-11 23:39 lib/python3.5/test/__pycache__/test_zipfile64.cpython-35.pyc
-rw-rw-r-- travis/staff  49911 2017-01-11 23:38 lib/python3.5/__pycache__/zipfile.cpython-35.pyc
-rw-rw-r-- travis/staff  76983 2017-01-11 23:38 lib/python3.5/test/__pycache__/test_zipfile.cpython-35.pyc
```
